File: integration_tests/workspaces/go/callchain.go
package main

import "fmt"

// LeafFunction is at the bottom of the call chain — called but calls nothing interesting
func LeafFunction() string {
	return "leaf result"
}

// MiddleFunction calls LeafFunction and is called by EntryPoint
func MiddleFunction() string {
	result := LeafFunction()
	return fmt.Sprintf("middle: %s", result)
}

// EntryPoint is the top of the call chain — calls MiddleFunction
func EntryPoint() {
	msg := MiddleFunction()
	fmt.Println(msg)
}

// AnotherCaller also calls MiddleFunction, giving it two incoming callers
func AnotherCaller() {
	fmt.Println(MiddleFunction())
}

File: integration_tests/workspaces/clangd/src/callchain.cpp
#include <iostream>
#include <string>

std::string leafFunction() {
    return "leaf result";
}

std::string middleFunction() {
    std::string result = leafFunction();
    return "middle: " + result;
}

void entryPoint() {
    std::string msg = middleFunction();
    std::cout << msg << std::endl;
}

void anotherCaller() {
    std::cout << middleFunction() << std::endl;
}
